sdcm/sct_events.py

```diff
@@ -113,6 +143,20 @@ def publish_event(self, event):
         with open(self.raw_events_filename, 'a+') as log_file:
             log_file.write(event.to_json() + '\n')
         socket.close()
+        return True
+
+    def publish_event_guaranteed(self, event, timeout=5, republish=3, reuse_socket=True):
```

Comment:
I prefer for the simplicity of the code use retrying decorator and not to expose `timeout` and `republish`:

```
@retrying(n=3, sleep_time=5, allowed_exceptions=SocketNotReady)
def publish_event_guaranteed(event, reuse_socket=True):
    client_socket = self.get_client_socket(reuse_socket=reuse_socket)
    self.publish_event(event)
    if not client_socket.poll(timeout=1):
       raise SocketNotReady()
    received_event = client_socket.recv_pyobj()
    if event == received_event:
        return True
    return False
```

Reply:
use retrying in other methods too

Reply:
Done